tests/consent/api/test_views.py

```diff
@@ -0,0 +1,452 @@
+            400
+        ),
+        # Missing `enterprise_customer` input.
+        (
+            factories.EnterpriseCourseEnrollmentFactory,
+            [{
+                'enterprise_customer_user__user_id': TEST_USER_ID,
+                'enterprise_customer_user__enterprise_customer__uuid': TEST_UUID,
+                'consent_granted': True,
+                'course_id': TEST_COURSE,
+            }],
+            {
+                'username': TEST_USERNAME,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': None,
+                'course_id': TEST_COURSE,
+            },
+            400
+        ),
+        # No consent in an enterprise course enrollment.
+        # Expect new consent resource to be created with consent provided.
+        (
+            factories.EnterpriseCustomerUserFactory,
+            [{
+                'user_id': TEST_USER_ID,
+                'enterprise_customer__uuid': TEST_UUID
+            }],
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
+            },
+            200
+        ),
+        # Consent already given for an enterprise course enrollment.
+        # Expect nothing to change.
+        (
+            factories.EnterpriseCourseEnrollmentFactory,
+            [{
+                'enterprise_customer_user__user_id': TEST_USER_ID,
+                'enterprise_customer_user__enterprise_customer__uuid': TEST_UUID,
+                'consent_granted': True,
+                'course_id': TEST_COURSE,
+            }],
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
+            },
+            200
+        ),
+        # Consent not given for an enterprise course enrollment.
+        # Expect consent to be given.
+        (
+            factories.EnterpriseCourseEnrollmentFactory,
+            [{
+                'enterprise_customer_user__user_id': TEST_USER_ID,
+                'enterprise_customer_user__enterprise_customer__uuid': TEST_UUID,
+                'consent_granted': False,
+                'course_id': TEST_COURSE,
+            }],
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
```

Comment:
So this is basically because I haven't actually made `POST` return any useful information. This is something I probably should have asked/discussed earlier: should `POST` return the following 2 extra keys:

```json
{
    "created": boolean,
    "consent_provided": boolean,
}
```

In my opinion, if `POST` returns a 200 status code, then it's obvious that `'consent_provided'` is going to be `True`, because we were able to successfully perform the only function relevant to `POST` here (provide consent), or create the resource and then perform that function. And if it returns 404, then `False`, because the request format was incorrect and we couldn't identify nor create the resource and provide consent. So do we need to return it? Either way, what do you reckon we should be returning to indicate state besides these?